core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -321,6 +331,22 @@ private void decoratePillarWithClouderaManagerDatabase(Cluster cluster, Map<Stri
                 new SaltPillarProperties("/cloudera-manager/database.sls", singletonMap("cloudera-manager", singletonMap("database", rdsView))));
     }
 
+    private void decoratePillarWithClouderaManagerLicense(Long stackId, Map<String, SaltPillarProperties> servicePillar)
+            throws CloudbreakOrchestratorFailedException {
+        String userCrn = stackService.get(stackId).getCreator().getUserCrn();
+
+        if (umsClient.isUmsUsable(userCrn)) {
```

Comment:
It would be better if we could hide this license retrieval to a service and it would retrieve only the license or a license for trial.

Reply:
There is no license for trial, unfortunately -- the trial is an API call (what you mentioned above), which makes this hard to do generally.

Reply:
My interpretation (and correct me if I'm wrong) is that we have to license CM at a different level of abstraction than enabling trial. We can enable a license via an API call as well, but it requires a restart of Cloudera Manager, which Altus would very much like to avoid.